adr/002-backend-data-initialization.md

```diff
@@ -0,0 +1,157 @@
+4. **Provide an “InitialInstanceData” bundle when opening an existing instance.**
+   When backend detects an instance context, it embeds (or exposes via single endpoint) the static-at-boot instance data adjacent to `AppData`, so the client does not immediately refetch/invalidate. Further interactions use targeted APIs. ([GitHub][1])
+5. **Remove RuleHandler/RuleConfiguration in this major.**
+   Do not migrate them to the new boot flow (handled by separate task).
+
+In scope: catalog and migrate all initial queries across stateless, stateless-anon, and stateful apps. Out of scope: data that must be updated _after_ boot.
+
+```mermaid
+graph LR
+A[User's Browser]:::actor
+
+subgraph S1["Before — frontend orchestrates boot"]
+A --> FFE[app-frontend]
+FFE -->|fetch on mount| Q1[GET /texts]
+FFE -->|fetch on mount| Q2[GET /app-metadata]
+FFE -->|fetch on mount| Q3[GET /profile/party]
+FFE -->|fetch on mount| Q4[GET /instances?active=true]
+FFE -->|fetch on mount| Q5[GET /orgs]
+FFE -->|fetch on mount| Q6[GET /ui-config]
+FFE -.-> FX[Complex provider/effect graph]
+end
+
+subgraph S2["After — backend assembles initial payload"]
+A --> BE[app-backend HomeController]
+BE -->|orchestrates| P1[(Platform APIs)]
+P1 --> BE
+BE -->|Bootstrap HTML + AppData + InitialInstanceData| A
+A --> FFE2[app-frontend hydrate & render]
+FFE2 -->|reads global variable| QC[global state variables]
+FFE2 -->|targeted fetch on demand| P2[(Platform APIs)]
+end
+
+classDef actor fill:#eef,stroke:#77f,color:#000
+```
+
+## Rationale
+
+- **Performance:** Fewer requests and no waterfall; better TTFB/TTI by shipping what the app needs up front.
+- **Simplicity:** One place orchestrates initial data; frontend boot becomes deterministic.
+- **Reliability:** Backend can decide what to load based on route/instance/party and enforce pagination/limits.
+
+## Considered Options
+
+1. **Status quo (frontend-orchestrated queries).**
+   Rejected: retains complexity and network overhead.
+2. **Bulk endpoints.**
+   Rejected: increases backend complexity without clear benefit vs app-backend controller.
+
+## Technical Notes (non-normative)
+
+- **Bootstrap contract**
+
+  ```html
+  <script
+    id="__appdata"
+    type="application/json"
+  >
+    { ... AppData ... }
+  </script>
+  <script
+    id="__initialInstanceData"
+    type="application/json"
+  >
+    { ... } <!-- present only when opening an instance -->
+  </script>
+  <script>
+    window.AltinnAppData = JSON.parse(document.getElementById('__appdata').textContent);
+  </script>
+  <script>
+    window.AltinnInitialInstanceData = JSON.parse(
+      document.getElementById('__initialInstanceData')?.textContent || 'null',
+    );
+  </script>
+  ```
+
+- **Routing:**
+  - Backend maps `/` `/instances/:id/*` `/tasks/:taskId` etc. to a single `HomeController` action that embeds the correct initial payload and serves static assets.
```

Comment:
Is it actually possible to make this work with our current API routes? E.g. we have `{org}/{app}/instances/{instanceOwnerPartyId:int}/{instanceGuid:guid}` as a route in `InstancesController`. Do we get a collision if we remove the hash? Or was the frontend route `instance` (singular)?